implementation

berkebakkalogluosf opened the pull request.

berkebakkalogluosf committed:
implementation

romanbychkoosf commented on WeatherInfoWidget/force-app/main/default/classes/WeatherCalloutExtension.cls:
```diff
@@ -0,0 +1,46 @@
+        request.setEndpoint('https://api.openweathermap.org/data/2.5/weather?q=' + account.BillingCity + '&appid=' + API_KEY);
+        request.setMethod('GET');
+        
+        response = http.send(request);
```
http requests have to be in try/catch also.

romanbychkoosf commented on WeatherInfoWidget/force-app/main/default/classes/WeatherCalloutExtension.cls:
```diff
@@ -0,0 +1,46 @@
+        
+        response = http.send(request);
+        
+        if (response.getStatusCode() == 200) {
```
I suggest to add a bit more conditions here
if status >= 200 AND < 300 --- OK
if status >=400 AND < 500 --- bad request
if status >=500 and < 600 --- some server error

try to output different error messages on a screen. You can use <apex:pageMessages /> in VF page

berkebakkalogluosf committed:
minor changes

- put the http request to a try/catch block
- checked for additional status codes and sent the error codes to the vf page
